File: 2022/14/javascript/main.mjs
import { loadMap, sim } from "./map.mjs";

// --- Constants ---
const INPUT_FILE = "../input/input.txt";

// --- Read and parse the input file ---
let map = await loadMap(INPUT_FILE);

// --- Puzzle 1 ---
sim(map, 1);
let total = map.countSandTiles();
console.log("1. Total units of sand:", total.toLocaleString());

// --- Puzzle 2 ---
sim(map, 2);
total = map.countSandTiles();
console.log("2. Total units of sand:", total.toLocaleString());

File: 2022/14/python/main.py
from enum import Enum
from lib import Map

# --- Constants --
INPUT_FILE = "../input/input.txt"

# --- Read and parse the input file ---
_map = Map(INPUT_FILE)

# --- Puzzle 1 ---
_map.sim(1)
total = _map.count_sand()
print(f"1. Total units of sand: {total:,d}")

# --- Puzzle 2 ---
_map.sim(2)
total = _map.count_sand()
print(f"2. Total units of sand: {total:,d}")
